frontend/src/app/main/ui/measurements.cljs

```diff
@@ -308,192 +316,552 @@
-                                     :pill-data (assoc pill-data
-                                                       :x (- x1 (+ pill-width pill-separation))
-                                                       :y (if (:flip-y frame)
-                                                            (+ y1 (/ (- (:p3 padding) pill-height) 2))
-                                                            (+ (- y2 (:p3 padding)) (/ (- (:p3 padding) pill-height) 2)))
-                                                       :value (:p3 padding))
-                                     :hover?  (or hover-all? hover-v? (= @hover :p3))
-                                     :selected? (:p3 paddings-selected)}
-
-                                    {:frame-id frame-id
-                                     :zoom zoom
-                                     :hover-all? hover-all?
-                                     :hover-v? hover-v?
-                                     :hover-h? hover-h?
-                                     :padding-num :p4
-                                     :padding padding
-                                     :on-mouse-enter (partial on-mouse-enter :p4)
-                                     :on-mouse-leave on-mouse-leave
-                                     :rect-data {:x (if (:flip-x frame) (- x2 (:p4 padding)) x1)
+                                            :p4 {:x (if (:flip-x frame) (- x2 (:p4 padding)) x1)
                                                  :y y1
                                                  :width (:p4 padding)
                                                  :height height
                                                  :initial-value (:p4 padding)
                                                  :resize-type (if (:flip-x frame) :right :left)
                                                  :resize-axis :x
-                                                 :resize-negate? (:p4 negate)}
-                                     :pill-data (assoc pill-data
-                                                       :x (if (:flip-x frame)
-                                                            (+ (- x2 (:p4 padding)) (/ (- (:p4 padding) pill-width) 2))
-                                                            (+ x1 (/ (- (:p4 padding) pill-width) 2)))
-                                                       :y (- y1 (+ pill-height pill-separation))
-                                                       :value (:p4 padding))
-                                     :hover?  (or hover-all? hover-h? (= @hover :p4))
-                                     :selected? (:p4 paddings-selected)}]]
+                                                 :resize-negate? (:p4 negate)}}]
 
     [:g.paddings {:pointer-events "visible"}
-     (for [data padding-display-data]
-       [:& padding-display data])]))
+     (for [[padding-num rect-data] padding-rect-data]
+       [:& padding-display {:frame-id frame-id
```

Comment:
when you create components in a loop you need to add `key="...."` otherwise the console will show a warning (and it's a best practice).